src/Workspaces/MSBuild/BuildHost/MSBuild/ProjectFile/ProjectFile.cs

```diff
@@ -274,7 +274,7 @@ private ImmutableArray<string> GetRelativeFolders(MSB.Framework.ITaskItem docume
             {
                 var filePath = documentItem.ItemSpec;
                 var relativePath = PathUtilities.GetDirectoryName(PathUtilities.GetRelativePath(_projectDirectory, filePath));
-                var folders = relativePath == null ? [] : relativePath.Split(PathUtilities.DirectorySeparatorChar, PathUtilities.AltDirectorySeparatorChar).ToImmutableArray();
+                var folders = relativePath == null ? [] : relativePath.Split([PathUtilities.DirectorySeparatorChar, PathUtilities.AltDirectorySeparatorChar], StringSplitOptions.RemoveEmptyEntries).ToImmutableArray();
```

Comment:
For files in the same folder as the project file, `PathUtilities.GetRelativePath(_projectDirectory, filePath)` would always return `filePath`. Meaning `PathUtilities.GetDirectoryName(...)` would return `""`. Since we were not removing empty entries the resultant return would be `[""]`.

Reply:
Can we have a test for this one?